flh_opt/api_opt.py

```diff
@@ -154,26 +153,27 @@ def add_storage(n: Network, input_data: dict, name: str, bus: str) -> None:
     add_storage(n, input_data, "H2_STR", "H2")
 
     # add RE profiles:
-    res_profiles = pd.DataFrame()
     for g in input_data["RES"]:
         process_code = g["PROCESS_CODE"]
         if len(input_data["RES"]) > 1:
             re_location = "RES_HYBR"
         else:
             re_location = process_code
-        res_profiles[process_code] = get_profiles(
+        res_profiles, weights = get_profiles_and_weights(
             source_region_code=input_data["SOURCE_REGION_CODE"],
-            process_code=process_code,
             re_location=re_location,
             selection=range(0, 48),  # TODO: make this a function parameter?
-        )["specific_generation"]
+        )
 
     # define snapshots:
     n.snapshots = res_profiles.index
```

Comment:
`res_profiles` currently has a MultiIndex with names `["period_id", "TimeStep"]`. I am not sure if that is okay for the pypsa network.

Reply:
i fixed this.